src/frontend/Komponenter/Personoversikt/RevurderingModal.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from 'react';
+import { Behandlingstype } from '../../App/typer/behandlingstype';
+import {
+    Behandlingsårsak,
+    behandlingsårsaker,
+    behandlingsårsakTilTekst,
+} from '../../App/typer/Behandlingsårsak';
+import { Flatknapp, Hovedknapp } from 'nav-frontend-knapper';
+import { AlertStripeFeil } from 'nav-frontend-alertstriper';
+import UIModalWrapper from '../../Felles/Modal/UIModalWrapper';
+import styled from 'styled-components';
+import { Select } from 'nav-frontend-skjema';
+import { FamilieDatovelger } from '@navikt/familie-form-elements';
+import { Ressurs, RessursStatus } from '../../App/typer/ressurs';
+import { useApp } from '../../App/context/AppContext';
+import { RevurderingInnhold } from '../../App/typer/revurderingstype';
+
+const StyledSelect = styled(Select)`
+    margin-top: 2rem;
+`;
+
+const KnappeWrapper = styled.div`
+    margin-top: 16rem;
+    margin-bottom: 1rem;
+`;
+
+const StyledFamilieDatovelgder = styled(FamilieDatovelger)`
+    margin-top: 2rem;
+`;
+
+const StyledHovedknapp = styled(Hovedknapp)`
+    margin-right: 1rem;
+`;
+
+interface IProps {
+    visModal: boolean;
+    settVisModal: (bool: boolean) => void;
+    fagsakId: string;
+    hentFagsak: () => void;
+    settSkalNavigereTilBehandlingen: (bool: boolean) => void;
+    settKanStarteRevurdering: (bool: boolean) => void;
+}
+
+const RevurderingsModal: React.FunctionComponent<IProps> = ({
+    visModal,
+    settVisModal,
+    fagsakId,
+    hentFagsak,
+    settSkalNavigereTilBehandlingen,
+    settKanStarteRevurdering,
+}) => {
+    const [feilmeldingModal, settFeilmeldingModal] = useState<string>();
+    const [valgtBehandlingstype, settValgtBehandlingstype] = useState<Behandlingstype>(
+        Behandlingstype.REVURDERING
+    );
+    const [valgtBehandlingsårsak, settValgtBehandlingsårsak] = useState<Behandlingsårsak>();
+    const [valgtDato, settValgtDato] = useState<string>();
+
+    const validerKanStarteRevurdering = (): boolean => {
+        return !!(valgtBehandlingstype && valgtBehandlingsårsak && valgtDato);
+    };
+    const { axiosRequest } = useApp();
+
+    const opprettRevurdering = () => {
+        settFeilmeldingModal('');
+        if (validerKanStarteRevurdering()) {
+            settKanStarteRevurdering(false);
+            settVisModal(false);
+            const revurderingInnhold: RevurderingInnhold = {
+                fagsakId: fagsakId,
+                behandlingsårsak: valgtBehandlingsårsak as Behandlingsårsak,
+                kravMottat: valgtDato as string,
+            };
+            axiosRequest<Ressurs<void>, RevurderingInnhold>({
+                method: 'POST',
+                url: `/familie-ef-sak/api/revurdering/${fagsakId}`,
+                data: revurderingInnhold,
+            }).then((response) => {
+                if (response.status === RessursStatus.SUKSESS) {
+                    settSkalNavigereTilBehandlingen(true);
```

Comment:
Får vi tilbake behandlingId i responsen? Kan vi isåfall bruke useHistory() direkte på den?

Reply:
isåfall usikker på om vi trenger `hentFagsak()` hvis vi løser redirecten direkte? 🤔